### ISSUE TITLE
t/v3.t: test failures

### ISSUE THREAD

**Issue by jkeenan** [jhthorsen/mojolicious-plugin-openapi#101]
When this module was tested against perl-5.29.7 on FreeBSD-11.2 today, test failures were observed in t/v3.t.  The output is over 3200 lines long; here is an excerpt:
```
$ head -80 t-v3.t-output.txt

#   Failed test 'valid openapi v3 schema'
#   at t/v3.t line 28.
# Undefined subroutine &JSON::Validator::Formats::uri called at /home/jkeenan/var/tad/testing/perl-5.29.7/lib/site_perl/5.29.7/JSON/Validator/Formats.pm line 100.
[2019-01-20 17:37:39.04886] [22880] [debug] GET "/pets" (359b2679)
[2019-01-20 17:37:39.04950] [22880] [debug] Routing to a callback
[2019-01-20 17:37:39.05026] [22880] [error] Can't locate object method "openapi" via package "Mojolicious::Controller" at t/v3.t line 16.

[2019-01-20 17:37:39.05174] [22880] [debug] Template "exception.development.html.ep" not found
[2019-01-20 17:37:39.05198] [22880] [debug] Template "exception.html.ep" not found
[2019-01-20 17:37:39.05233] [22880] [debug] Rendering template "mojo/debug.html.ep"
[2019-01-20 17:37:39.06768] [22880] [debug] Your secret passphrase needs to be changed
[2019-01-20 17:37:39.06992] [22880] [debug] 500 Internal Server Error (0.021056s, 47.492/s)

#   Failed test '400 Bad Request'
#   at t/v3.t line 31.
#          got: '500'
#     expected: '400'

#   Failed test 'exact match for JSON Pointer "/errors/0/message"'
#   at t/v3.t line 31.
#          got: undef
#     expected: 'Expected integer - got string.'
[2019-01-20 17:37:39.07445] [22880] [debug] GET "/pets" (22a61313)
[2019-01-20 17:37:39.07469] [22880] [debug] Routing to a callback
[2019-01-20 17:37:39.07542] [22880] [error] Can't locate object method "openapi" via package "Mojolicious::Controller" at t/v3.t line 16.

[2019-01-20 17:37:39.07577] [22880] [debug] Template "exception.development.html.ep" not found
[2019-01-20 17:37:39.07600] [22880] [debug] Template "exception.html.ep" not found
[2019-01-20 17:37:39.07626] [22880] [debug] Rendering cached template "mojo/debug.html.ep"
[2019-01-20 17:37:39.08046] [22880] [debug] 500 Internal Server Error (0.005999s, 166.694/s)

#   Failed test 'exact match for JSON Pointer "/errors/0/message"'
#   at t/v3.t line 35.
#          got: undef
#     expected: 'No responses rules defined for type not/supported.'
[2019-01-20 17:37:39.08385] [22880] [debug] GET "/pets" (9aabdbe1)
[2019-01-20 17:37:39.08417] [22880] [debug] Routing to a callback
[2019-01-20 17:37:39.08485] [22880] [error] Can't locate object method "openapi" via package "Mojolicious::Controller" at t/v3.t line 16.

[2019-01-20 17:37:39.08514] [22880] [debug] Template "exception.development.html.ep" not found
[2019-01-20 17:37:39.08538] [22880] [debug] Template "exception.html.ep" not found
[2019-01-20 17:37:39.08553] [22880] [debug] Rendering cached template "mojo/debug.html.ep"
[2019-01-20 17:37:39.08972] [22880] [debug] 500 Internal Server Error (0.005853s, 170.853/s)

#   Failed test 'exact match for JSON Pointer "/errors/0/message"'
#   at t/v3.t line 38.
#          got: undef
#     expected: 'Expected array - got object.'
[2019-01-20 17:37:39.09304] [22880] [debug] GET "/pets" (e480dbfa)
[2019-01-20 17:37:39.09329] [22880] [debug] Routing to a callback
[2019-01-20 17:37:39.09398] [22880] [error] Can't locate object method "openapi" via package "Mojolicious::Controller" at t/v3.t line 16.

[2019-01-20 17:37:39.09428] [22880] [debug] Template "exception.development.html.ep" not found
[2019-01-20 17:37:39.09451] [22880] [debug] Template "exception.html.ep" not found
[2019-01-20 17:37:39.09472] [22880] [debug] Rendering cached template "mojo/debug.html.ep"
[2019-01-20 17:37:39.09891] [22880] [debug] 500 Internal Server Error (0.005861s, 170.619/s)

#   Failed test '200 OK'
#   at t/v3.t line 41.
#          got: '500'
#     expected: '200'

#   Failed test 'exact match for content'
#   at t/v3.t line 41.
#          got: '<!DOCTYPE html>
# <!-- Request ID: e480dbfa -->
# <html>
#   <head>
#     <title>Server error (development mode)</title>
#     <meta http-equiv="Pragma" content="no-cache">
#     <meta http-equiv="Expires" content="-1">
#     <script src="/mojo/jquery/jquery.js"></script>
#     <script src="/mojo/prettify/run_prettify.js"></script>
#     <link href="/mojo/prettify/prettify-mojo-dark.css" rel="stylesheet">
#     <style>
#       a img {
#         border: 0;
#       }
#       body {
```

```
$ tail -30 t-v3.t-output.txt
ok 9 - 500 Internal Server Error
not ok 10 - exact match for JSON Pointer "/errors/0/message"
ok 11 - GET /pets?limit=10
not ok 12 - 200 OK
not ok 13 - exact match for content
ok 14 - GET /pets?limit=10
not ok 15 - 200 OK
not ok 16 - exact match for content
ok 17 - POST /pets
not ok 18 - 400 Bad Request
not ok 19 - exact match for JSON Pointer "/errors/0/message"
not ok 20 - exact match for JSON Pointer "/errors/1/message"
ok 21 - POST /pets
not ok 22 - 201 Created
not ok 23 - exact match for content
ok 24 - POST /pets
not ok 25 - 201 Created
not ok 26 - exact match for content
1..26
Dubious, test returned 16 (wstat 4096, 0x1000)
Failed 16/26 subtests 

Test Summary Report
-------------------
t/v3.t (Wstat: 4096 Tests: 26 Failed: 16)
  Failed tests:  1, 3-4, 7, 10, 12-13, 15-16, 18-20, 22-23
                25-26
  Non-zero exit status: 16
Files=1, Tests=26,  1 wallclock secs ( 0.04 usr  0.03 sys +  0.45 cusr  0.14 csys =  0.66 CPU)
Result: FAIL
```
Possibly related to https://github.com/mojolicious/json-validator/issues/144

**Comment by jhthorsen** [comment]
Which version are you testing?

**Comment by jkeenan** [comment]
On 1/20/19 7:31 PM, Jan Henning Thorsen wrote:
> Which version are you testing?
> 

Mojolicious-Plugin-OpenAPI-2.08

**Comment by jhthorsen** [comment]
I updated my comment: https://github.com/jhthorsen/mojolicious-plugin-openapi/issues/101#issuecomment-455917348

**Comment by jkeenan** [comment]
JSON-Validator-3.03
Mojolicious-Plugin-OpenAPI-2.08

**Comment by jhthorsen** [comment]
I think I see the bug now, but this should be fixed in 3.04.
